todo_app/app.py

```diff
@@ -1,11 +1,21 @@
-from flask import Flask
+from flask import Flask, render_template, request
 
 from todo_app.flask_config import Config
+from todo_app.data.session_items import add_item, get_items
+
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
 
-@app.route('/')
+@app.route('/', methods=['GET', 'POST'])
 def index():
-    return 'Hello World!'
+    if (request.method == 'GET'):
+        return render_template('index.html', items = get_items())
+
+    if (request.method == 'POST'):
+        title = request.form.get('title')
+        add_item(title)
+        return render_template('index.html', items = get_items())
```

Comment:
You have two totally separate cases here - the user is trying to view the homepage and the user is trying to add a new item. So instead of switching between them with `if` clauses, define two routes:

```python
@app.route('/', methods=['GET'])
def index():
    ...
    
@app.route('/', methods=['POST'])
def add_item_route():
    ...
```

Secondly, instead of directly rendering the index page after adding an item, it'd be a bit nicer to return a redirect response (as the PDF suggests). This will for example prevent issues when the user tries to refresh the page after submitting the form.

Reply:
done